leopard8.go

```diff
@@ -463,19 +467,19 @@ func (r *leopardFF8) reconstruct(shards [][]byte, recoverAll bool) error {
 	for i := 0; i < r.parityShards; i++ {
 		if len(shards[i+r.dataShards]) == 0 {
 			errLocs[i] = 1
-			if LEO_ERROR_BITFIELD_OPT && recoverAll {
+			if LEO_ERROR_BITFIELD_OPT && (recoverAll || (required != nil && required[i])) {
 				errorBits.set(i)
 			}
 		}
 	}
 	for i := r.parityShards; i < m; i++ {
 		errLocs[i] = 1
-		if LEO_ERROR_BITFIELD_OPT && recoverAll {
+		if LEO_ERROR_BITFIELD_OPT && (recoverAll || (required != nil && required[i])) {
 			errorBits.set(i)
 		}
 	}
 	for i := 0; i < r.dataShards; i++ {
-		if len(shards[i]) == 0 {
+		if len(shards[i]) == 0 /*&& (recoverAll || (required != nil && required[i]))*/ {
```

Comment:
same as in leopard.go:

https://github.com/klauspost/reedsolomon/pull/274/files#r1511442291